examples/applications/plot_2d_constrained.py

```diff
@@ -0,0 +1,126 @@
+"""
+======================================
+Constrained (2D)
+======================================
+
+How to run a basic 2D constrained proximal primal-dual solver. 
+We consider the canonical problem :math:`y = x + n` where :math:`n \sim \mathcal{N}`. 
+This inverse problem can be solved via the constrained optimisation 
+
+.. math::
+
+    \min_x ( ||\Psi^{\dagger} x||_1 ) \quad s.t. \quad ||(x-y)||^2_2 \leq \epsilon
+
+where :math:`x \in \mathbb{R}^2` is an a priori ground truth 2D signal, :math:`y \in \mathbb{R}^2` 
+are simulated noisy observations, and :math:`\epsilon` is the radius of the :math:`\ell_2`-ball, i.e. 
+a likelihood isocontour. Before we begin, we need to import ``optimusprimal`` and some example 
+specific packages
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+from skimage import data
+from scipy.signal import resample
+
+import optimusprimal.primal_dual as primal_dual
+import optimusprimal.grad_operators as grad_operators
+import optimusprimal.linear_operators as linear_operators
+import optimusprimal.prox_operators as prox_operators
+
+##############################################################################
+# First, we need to define some heuristics for the solver, these include:
+#       - tol: convergence criteria for the iterations
+#       - iter: maximum number of iterations
+#       - update_iter: iterations between logging iteration diagnostics
+#       - record_iters: whether to record the full diagnostic information
+
+options = {"tol": 1e-5, "iter": 5000, "update_iter": 50, "record_iters": False}
+
+##############################################################################
+# Next, we simulate a standard de-noising setting by contaminating a known
+# signal :math:`x`` with some Gaussianly distributed noise. Note that for simplicity the
+# measurement operator here is taken to be the identity operator.
+
+ISNR = 25.0                                            # Input signal to noise ratio
+sigma = 10 ** (-ISNR / 20.0)                           # Noise standard deviation
+res = 128                                              # Resolution we want to work with
+epsilon = np.sqrt(res ** 2 + 2.0 * res) * sigma        # Radius of l2-ball
+
+
+x = data.camera()                                       # Canonical cameraman image
```

Comment:
We could use scipy.misc.ascent() so we do not need to add skimage to requirements (if we are not using it anywhere else)

Reply:
Great point, updated in the examples now.